### PR TITLE
Fix Robotic Workforce to work with Ares cards

### PR DESCRIPTION
**Pull request by Lynesth**
Please check the PR commit by commit.
The first one contains the actual changes.
The second one is simply formatting and reordering things around.
Tests inspired by @Starcalc.

### CODE REVIEW

File: src/cards/base/RoboticWorkforce.ts [hunk 1]
```diff
@@ -196,27 +211,40 @@ export class RoboticWorkforce implements IProjectCard {
...
           if (bonusResource !== undefined && bonusResource === Resources.STEEL) {
             this.miningSteelProduction++;
           }
           if (bonusResource !== undefined && bonusResource === Resources.TITANIUM) {
             this.miningTitaniumProduction++;
           }
+          break;
+
+        case CardName.SOLAR_FARM:
+          const solarFarmSpace = game.board.getSpaceByTileCard(CardName.SOLAR_FARM);
```
Comment (kberg) [inline src/cards/base/RoboticWorkforce.ts]: This is a clever idea. I would like to add this to the Ares wiki page for how this game deals with outliers. Am I reading it right that what you're doing here is adding the number of energy based on the number of plants the tile was placed on?

Comment (kberg) [reply src/cards/base/RoboticWorkforce.ts]: Same comment applies to Mining Area and Mining Rights.

**Review by kberg — APPROVED** [review]
Thanks for splitting out the reformat.

File: src/cards/base/RoboticWorkforce.ts (hunk 1 above)
Comment (Lynesth) [reply src/cards/base/RoboticWorkforce.ts]: Yes that's what it does for Solar Farm.

For Mining Area and Rights something was already in for those two, that saves the production you get directly on the card (in the `bonusResource` member). See the `play` method of `src/cards/base/MiningCard.ts` for details.

**Pull request merged**